### PR TITLE
add premake build file.

### PR DESCRIPTION
**Pull request by tvandijck**

### CODE REVIEW

**Comment by tvandijck** [comment]
https://github.com/premake/premake-core

File: src/longtail.c [hunk 1]
```diff
@@ -1278,7 +1278,7 @@ struct HashJob
 {
     struct Longtail_StorageAPI* m_StorageAPI;
     struct Longtail_HashAPI* m_HashAPI;
-    TLongtail_Hash* m_PathHash;
```
Comment (DanEngelbrecht) [inline src/longtail.c:1281]: Changes in this file can be reverted as a different solution is in place.

**Review by DanEngelbrecht — CHANGES_REQUESTED** [review]
path hash changes can be reverted otherwise good stuff!

File: cmd/ext/kgflags.h [hunk 2]
```diff
@@ -373,15 +373,15 @@ bool kgflags_parse(int argc, char **argv) {
         _kgflags_parse_flag(flag, prefix_no);
     }
 
-    _kgflags_assign_default_values();
```
Comment (DanEngelbrecht) [inline cmd/ext/kgflags.h:376]: Not sure about these changes either, this is in a third-party repo, should probably start by updating that to latest.
Also, by moving _kgflags_assign_default_values the behavior is changed I guess?
Flags that are required but have a default value can't fall back to default now, right? You *must* provide a value even if it has default.
Debatable weather a flag should be able to be set both required and have a default, but the behavior of is changed with this, right?

Comment (tvandijck) [reply cmd/ext/kgflags.h:376]: Yeah, I'm not sure what the deal is here.... but for me defaults didn't work at all.. So I just tried a few things until it works "for me"... I think in general this library is just really bad... there is better out there, or even just doing a custom roll would be easier to understand.

**Review by DanEngelbrecht — APPROVED** [review]

**Pull request merged**